lib/tests/unit/modules/task_test.cc

```diff
@@ -206,6 +250,73 @@ TEST_CASE("Modules::Task::executeAction", "[modules][output]") {
         REQUIRE(response.output.std_err == "");
         REQUIRE(response.output.exitcode == 0);
     }
-}
 
+    SECTION("errors on download when no master-uri is provided") {
+        Modules::Task e_m { PXP_AGENT_BIN_PATH, TEMP_TASK_CACHE_DIR, {}, CA, CRT, KEY, SPOOL_DIR };
+        auto task_txt = (DATA_FORMAT % "\"0632\""
+                                     % "\"task\""
+                                     % "\"run\""
+                                     % "{\"task\": \"unparseable\", \"input\":{\"message\":\"hello\"}, "
+                                       "\"files\" : [{\"sha256\": \"some_sha\", \"filename\": \"some_file\"}]}").str();
+        PCPClient::ParsedChunks task_content {
+            lth_jc::JsonContainer(ENVELOPE_TXT),
+            lth_jc::JsonContainer(task_txt),
+            {},
+            0 };
+        ActionRequest request { RequestType::Blocking, task_content };
+        auto response = e_m.executeAction(request);
+
+        REQUIRE_FALSE(response.action_metadata.includes("results"));
+        REQUIRE_FALSE(response.action_metadata.get<bool>("results_are_valid"));
+        REQUIRE(response.action_metadata.includes("execution_error"));
+        REQUIRE(boost::contains(response.action_metadata.get<std::string>("execution_error"), "Cannot download task. No master-uris were provided"));
+    }
+
+    SECTION("errors when a download error occurs and removes the temporary file") {
+        Modules::Task e_m { PXP_AGENT_BIN_PATH, TEMP_TASK_CACHE_DIR, MASTER_URIS, CA, CRT, KEY, SPOOL_DIR };
+        auto cache = fs::path(TEMP_TASK_CACHE_DIR) / "some_sha";
+        std::string bad_path = "bad_path";
+        auto task_txt = (DATA_FORMAT % "\"0632\""
+                                     % "\"task\""
+                                     % "\"run\""
+                                     % "{\"task\": \"unparseable\", \"input\":{\"message\":\"hello\"}, "
+                                       "\"files\" : [{\"uri\": { \"path\": \"bad_path\", \"params\": { \"environment\": \"production\", \"code-id\": \"1234\" }}, \"sha256\": \"some_sha\", \"filename\": \"some_file\"}]}").str();
+        PCPClient::ParsedChunks task_content {
+            lth_jc::JsonContainer(ENVELOPE_TXT),
+            lth_jc::JsonContainer(task_txt),
+            {},
+            0 };
+        ActionRequest request { RequestType::Blocking, task_content };
+        auto response = e_m.executeAction(request);
+
+        REQUIRE_FALSE(response.action_metadata.includes("results"));
+        REQUIRE_FALSE(response.action_metadata.get<bool>("results_are_valid"));
+        REQUIRE(response.action_metadata.includes("execution_error"));
+        REQUIRE_THAT(response.action_metadata.get<std::string>("execution_error"), Catch::Contains("_mock_master_" + bad_path));
```

Comment:
This appears to be inconsistent with different versions of curl. On my laptop it passes, in Travis it fails because the url is single-quoted. We should probably make the test simply search for the bad path in the output.